pkg/network/server.go

```diff
@@ -611,7 +611,11 @@ func (s *Server) handleGetBlocksCmd(p Peer, gb *payload.GetBlocks) error {
 
 // handleGetBlockByIndexCmd processes the getblockbyindex request.
 func (s *Server) handleGetBlockByIndexCmd(p Peer, gbd *payload.GetBlockByIndex) error {
-	for i := gbd.IndexStart; i < gbd.IndexStart+uint32(gbd.Count); i++ {
+	count := gbd.Count
+	if gbd.Count < 0 || gbd.Count > payload.MaxHeadersAllowed {
+		count = payload.MaxHeadersAllowed
```

Comment:
In C# they use `InvPayload.MaxHashesCount` = 500 (see [link](https://github.com/neo-project/neo/blob/master/src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs#L192)), but in `(d *GetBlockByIndex) DecodeBinary` we have `payload.MaxHeadersAllowed` = 2000 (see also [C# implementation](https://github.com/neo-project/neo/blob/master/src/neo/Network/P2P/Payloads/GetBlockByIndexPayload.cs#L27)), perhaps it's a bug. I think that we should use the same constant to restrictthe number of blocks in both `DecodeBinary` and `handleGetBlockByIndexCmd`.

Reply:
I think first value is the amount of blocks that C# nodes can respond with, and the second is a receive limit. There is nothing wrong it those 2 being different (while the first is less or equal to the second).

Reply:
Especially when this payload is used for both requesting headers and blocks.

Reply:
See neo-project/neo#1813 also.